internal/pkg/aws/route53/route53.go

```diff
@@ -79,3 +165,28 @@ func matchesDomain(domain string) filterZoneFunc {
 		return domain == aws.StringValue(z.Name) || domain+"." == aws.StringValue(z.Name)
 	}
 }
+
+func cleanNSRecord(record string) string {
+	if !strings.HasSuffix(record, ".") {
+		return record
+	}
+	return record[:len(record)-1]
+}
+
+func isStrictSubset(subset, superset []string) bool {
+	if len(subset) > len(superset) {
+		return false
+	}
+
+	exists := make(map[string]struct{}, len(superset))
```

Comment:
nit: what do you think of naming it to something like this:
```suggestion
	isInSuperset := make(map[string]struct{}, len(superset))
```

Reply:
Renamed to `isMember`